src/pudl/extract/vcerare.py

```diff
@@ -0,0 +1,191 @@
+                {'year': 2009}
+                {'year_month': '2020-08'}
+
+        Returns:
+            string name of the CSV file
+        """
+        return f"{partition['year']}/{self._metadata.get_file_name(page, **partition)}"
+
+    def load_source(self, page: str, **partition: PartitionSelection) -> pd.DataFrame:
+        """Produce the dataframe object for the given partition.
+
+        Args:
+            page: pudl name for the dataset contents, eg "boiler_generator_assn" or
+                "data"
+            partition: partition to load. Examples:
+                {'year': 2009}
+                {'year_month': '2020-08'}
+
+        Returns:
+            pd.DataFrame instance containing CSV data
+        """
+        with (
+            self.ds.get_zipfile_resource(self._dataset_name, **partition) as zf,
+        ):
+            # Get list of file names in the zipfile
+            files = zf.namelist()
+            # Get the particular file of interest
+            file = next(
+                (x for x in files if self.source_filename(page, **partition) in x), None
+            )
+
+            # Read it in using pandas
+            # Set all dtypes except for the first unnamed hours column
+            # to be float32 to reduce memory on read-in
+            dtype_dict = defaultdict(lambda: np.float32)
+            dtype_dict["Unnamed: 0"] = (
+                "int"  # Set first unnamed column (hours) to be an integer.
+            )
+
+            df = pd.read_csv(BytesIO(zf.read(file)), dtype=dtype_dict)
+
+        return df
+
+    def process_raw(
+        self, df: pd.DataFrame, page: str, **partition: PartitionSelection
+    ) -> pd.DataFrame:
+        """Append report year to df to distinguish data from other years."""
+        self.cols_added.append("report_year")
+        selection = self._metadata._get_partition_selection(partition)
+        return df.assign(report_year=selection)
+
+    def validate(
+        self, df: pd.DataFrame, page: str, **partition: PartitionSelection
+    ) -> pd.DataFrame:
+        """Skip this step, as we aren't renaming any columns."""
+        return df
+
+    def combine(self, dfs: list[pd.DataFrame], page: str) -> pd.DataFrame:
+        """Concatenate dataframes into one, take any special steps for processing final page."""
+        df = pd.concat(dfs, sort=True, ignore_index=True)
+
+        return self.process_final_page(df, page)
+
+
+raw_vcerare__all_dfs = raw_df_factory(Extractor, name="vcerare")
+
+
+def raw_vcerare_asset_factory(part: str) -> AssetsDefinition:
+    """An asset factory for VCE hourly renewable generation profiles."""
+    asset_kwargs = {
+        "name": f"raw_vcerare__{part}",
+        "required_resource_keys": {"datastore", "dataset_settings"},
+    }
+
+    @asset(**asset_kwargs)
+    def _extract(context, raw_vcerare__all_dfs):
+        """Extract raw GridPath RA Toolkit renewable energy generation profiles.
```

Comment:
This seems like an accidental copy-over from the gridpath extractor.

Reply:
Sure is, good catch!